frontend/dev/script.ts

```diff
@@ -0,0 +1,223 @@
+let currentList: object[] = [];
+let currentPage: number = 1;
+const itemsPerPage: number = 5;
+
+const contentContainer: HTMLDivElement | null = document.querySelector('#content');
+const itemsContainer: HTMLUListElement | null = document.querySelector('#list');
+const pagination: HTMLDivElement | null = document.querySelector('#pagination');
+const currentPageSpan: HTMLSpanElement | null = document.querySelector('#num');
+const buttonNext: HTMLButtonElement | null = document.querySelector('#next');
+const buttonPrev: HTMLButtonElement | null = document.querySelector('#prev');
+
+type listItem = {
+  name: string;
+  descr: string;
+  url: string;
+  price: number;
+}
+
+if (buttonNext && buttonPrev) {
+  buttonNext.addEventListener('click', nextPage);
+  buttonPrev.addEventListener('click', prevPage);
+}
+
+async function getDataAndUpdatePage(): Promise<void> {
+  hideCatalog();
+  hidePagination();
+  showSpinner();
+
+  currentList = await getDataFromBE(currentPage);
+
+  if (currentList.length === 0 || currentList.length > itemsPerPage) {
+    errorOccurred();
+  } else {
+    updatePageNum();
+    updateCatalog();
+  }
+}
+
+getDataAndUpdatePage();
+updatePaginationButtons();
+
+async function getDataFromBE(page: number) {
+  try {
+    return (await fetch(`http://localhost:5500/api/sneakers/${page}`))
```

Comment:
domain name (localhost:5500) better save in variable. this is a value that can change